scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -952,6 +955,7 @@ def _response_values_validation(self, expected_response, received_response, resu
                 result.success(check_type, error_success.format(
                     name=expected_name, value=expected_value))
             else:
+                logging.warning(f"Response values validation mismatch: Expected Result: {expected_value} / Received Value: {received_value}")
```

Comment:
But doesn't all that get logged as part of the error_failure string already?

It would be useful to see typical before/after output here.  Because normally failures do in fact log the expected/received bits, in my experience....

Reply:
Hi @bzbarsky-apple, this is the log that is presented when validation fails:
`WARNING  | 2023-10-23 17:19:46.654 | app.test_engine.models.test_step:append_failure:55 | Test Failure: The test expectation "value == [{'GroupId': 259, 'Endpoints': ['1'], 'GroupName': 'Test Group'}]" is false`


The plan is to add more information by presenting what is used in comparison:
`WARNING  | 2023-10-23 17:19:46.446 | matter_yamltests.parser:_response_values_validation:952 | Response values validation mismatch: Expected Result: [{'GroupId': 259, 'Endpoints': ['1'], 'GroupName': 'Test Group'}] / Received Value: [{'GroupId': 259, 'Endpoints': [1], 'GroupName': 'Test Group', 'FabricIndex': 1}]`

Reply:
Hmm.  I'm quite sure that at least tests as run in CI do log the expected and actual values.  Is this about TH and it logging things differently somehow?

So for example, if I run Test_TC_SC_5_1 locally via the run_test_suite.py script, I get output like so:
```
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		    Expected: [
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                {
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                  "values": [
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                    {
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                      "value": [
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                        {
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                          "Endpoints": [
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                            1
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                          ],
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                          "GroupId": 259,
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                          "GroupName": "Test Group2"
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                        }
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                      ]
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                    }
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                  ]
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                }
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		              ]
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		    Received: [
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                 {
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                   "attribute": "GroupTable",
2023-10-23 23:08:54.479 ERROR   23:08:54.269 - TEST OUT  : 		                   "cluster": "GroupKeyManagement",
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                   "dataVersion": 978574612,
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                   "endpoint": 0,
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                   "value": [
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                     {
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                       "Endpoints": [
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                         1
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                       ],
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                       "FabricIndex": 1,
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                       "GroupId": 259,
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                       "GroupName": "Test Group"
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                     }
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                   ]
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		                 }
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		               ]
2023-10-23 23:08:54.480 ERROR   23:08:54.269 - TEST OUT  : 		***** Test Failure : The test expectation "value == [{'GroupId': 259, 'Endpoints': [1], 'GroupName': 'Test Group2'}]" is false
```
```